zcash_primitives/src/transaction/mod.rs

```diff
@@ -171,30 +184,48 @@ impl TxVersion {
         match self {
             TxVersion::Sprout(v) => *v >= 2u32,
             TxVersion::Overwinter | TxVersion::Sapling => true,
-            TxVersion::ZcashTxV5 => false,
+            TxVersion::Zip225 => false,
             #[cfg(feature = "zfuture")]
             TxVersion::ZFuture => true,
         }
     }
 
+    pub fn has_overwinter(&self) -> bool {
+        !matches!(self, TxVersion::Sprout(_))
+    }
+
     pub fn has_sapling(&self) -> bool {
         match self {
             TxVersion::Sprout(_) | TxVersion::Overwinter => false,
             TxVersion::Sapling => true,
-            TxVersion::ZcashTxV5 => true,
+            TxVersion::Zip225 => true,
+            #[cfg(feature = "zfuture")]
+            TxVersion::ZFuture => true,
+        }
+    }
+
+    pub fn has_orchard(&self) -> bool {
+        match self {
+            TxVersion::Sprout(_) | TxVersion::Overwinter | TxVersion::Sapling => false,
+            TxVersion::Zip225 => true,
             #[cfg(feature = "zfuture")]
             TxVersion::ZFuture => true,
         }
     }
 
+    #[cfg(feature = "zfuture")]
+    pub fn has_tze(&self) -> bool {
+        matches!(self, TxVersion::ZFuture)
+    }
+
     pub fn suggested_for_branch(consensus_branch_id: BranchId) -> Self {
         match consensus_branch_id {
             BranchId::Sprout => TxVersion::Sprout(2),
             BranchId::Overwinter => TxVersion::Overwinter,
             BranchId::Sapling | BranchId::Blossom | BranchId::Heartwood | BranchId::Canopy => {
                 TxVersion::Sapling
             }
-            BranchId::Nu5 => TxVersion::Sapling, //TEMPORARY WORKAROUND
+            BranchId::Nu5 => TxVersion::Zip225, //TEMPORARY WORKAROUND
```

Comment:
```suggestion
            BranchId::Nu5 => TxVersion::Zip225,
```

Reply:
What is this a workaround for?

Reply:
Oh, in an earlier PR before I'd actually implemented any ZIP-225 handling, I had reused the ZFuture constant, and for whatever reason accidentally forgot to remove the comment once TxVersion::Zip225 was actually available.